clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1471,6 +1487,7 @@ public void unsubscribe() {
             fetchBuffer.retainAll(Collections.emptySet());
             Timer timer = time.timer(defaultApiTimeoutMs);
             UnsubscribeEvent unsubscribeEvent = new UnsubscribeEvent(calculateDeadlineMs(timer));
+            backgroundEventProcessor.setIgnorePreviousBackgroundErrorEvent(timer.currentTimeMs());
```

Comment:
Hi @lianetm, thanks for the review. I updated a version to ignore `InvalidTopicException`.

I would like to check again. Do you think that we can make this function more reusable? Like, we can let developers set ignore exception type in the function, so if there are othere parts need to ignore something, they can just reuse the function. Or we don't need to over-engineering at this time? Thank you.

Reply:
I imagine we may end up in a similar situation with a TopicAuthorizationException but I would say we take it one at a time when we understand the issue (I don't have a clear one in mind with the auth path yet). Also, this is all very dependant on the api call so makes sense to target it separated like that (ignoring events is very delicate I would say, so I would avoid overdoing it)

Seeing like this (it's truly at the unsubscribe level where we can be specific about what we want to ignore, to make sure we're not ignoring more),  I wonder if we should consider just handling the InvalidTopic in a catch when processing background events after the unsubscribe? Seems simpler and clear to understand the reason for ignoring. Also I think it will be safer than having the current 2 phase mechanism, that requires that we "set ignore" -> do stuff -> "unset ignore" (easy to misuse and forget to flip it back). What do you think?

Reply:
Hi @lianetm, thanks for the suggestion. I update `processBackgroundEvents` function to accept a new argument `Predicate<Exception> ignoreErrorEventException`, so we don't need to do 2 phase mechanism in `BackgroundEventProcessor`.